### ISSUE TITLE
firstaction and lastaction do not check "hasErrors"

### ISSUE THREAD

**Issue by feimeng** [logrotate/logrotate#12]
When logrotate is run with "-f" option, the firstaction and lastaction will be run even there are errors detected. For example, if I do a "logrotate -f /path/to/lgorotate.conf" after logrotate has happened in the current hour, logrotate will not stomp the rotated file after it finds the file already exists. However, the lastaction script is still run because it did not check the "hasErros" variable. Is this intentional or a bug? The "-f" option will set numRotated no-zero.

**Comment by hanzz** [comment]
The problem with firstaction is that it's executed before the rotation begins. So when there are log files to rotate, firstaction is called, but later logrotate finds out that the log file with the same date extension exists and skips the rotation.

I could check for this particular error before running firstaction script, but there are bunch of other errors which cannot be detected like that. I therefore think the situation with "firstaction" execution is OK. During the firstaction, we just do know if the future rotation succeeds or not.

The lastaction is more problematic. Logrotate could stop calling that script and even the man-page suggests that by "lastaction is called .... only if at least one log is rotated". We actually have that info, so maybe it's really good thing to check for the errors there.

What worries me little bit that some people could use firstaction and lastaction to cleanup after each rotation without taking the care about rotation success. For those people, logrotate would break if lastaction would be called in case of rotation errror.

Anyway, I think for now I will:
a) Change the documentation for firstaction to mention that the script is called even when the rotation ends up with an error
b) Change the lastaction to be called only when the single log has been rotated
c) Change the lastaction documentation to mention that if rotation ends up with an error resulting in "no log rotated", then the script is not called.

**Comment by feimeng** [comment]
I agree that firstaction does not have enough knowledge whether the rotation will succeeds or not. Updating the documentation sounds good enough. For lastaction, I have patched out internal repo with the error check. Another subtle suggestion of documentation is to highlight the "-f" option will actually set the **numRotated** non-zero. If the user wants to use "-f", be aware of the implication of scripts.
